src/containerapp/azext_containerapp/custom.py

```diff
@@ -4297,3 +4306,195 @@ def delete_workload_profile(cmd, resource_group_name, env_name, workload_profile
+                    else:
+                        # Not based on image from mcr.microsoft.com/dotnet
+                        results.append(dict(targetContainerAppName=inspect_result["targetContainerAppName"], targetContainerAppEnvironmentName=inspect_result["targetContainerAppEnvironmentName"], targetResourceGroup=inspect_result["targetResourceGroup"], oldRunImage=inspect_result["remote_info"]["run_images"], newRunImage=None, id=None, reason=mcr_check_reason))
+    # Make sure that we clear the first line before showing the new output
+    print("\r                                   \r", end="", flush=True)
+    if show_all is False:
+        results = [result for result in results if result["id"] is not None]
+    if not results:
+        print("No container apps available to patch at this time. Use --show-all to show the container apps that cannot be patched.")
+        return
+    return results
+
+
+def patch_get_image_inspection(pack_exec_path, img, info_list):
+    if (img["imageName"].find("run-dotnet") != -1) and (img["imageName"].find("cbl-mariner") != -1):
+        inspect_result = {"remote_info": {"run_images": [{"name": "mcr.microsoft.com/oryx/builder:" + img["imageName"].split(":")[-1]}]}, "image_name": img["imageName"], "targetContainerName": img["targetContainerName"], "targetContainerAppName": img["targetContainerAppName"], "targetContainerAppEnvironmentName": img["targetContainerAppEnvironmentName"], "targetResourceGroup": img["targetResourceGroup"]}
+    else:
+        img_info = subprocess.Popen(pack_exec_path + " inspect-image " + img["imageName"] + " --output json", shell=True, stderr=subprocess.PIPE, stdout=subprocess.PIPE)
+        img_info_out, img_info_err = img_info.communicate()
+        if img_info_err.find(b"status code 401 Unauthorized") != -1 or img_info_err.find(b"unable to find image") != -1:
+            inspect_result = dict(remote_info=401, image_name=img["imageName"])
+        else:
+            inspect_result = json.loads(img_info_out)
+        inspect_result.update({"targetContainerName": img["targetContainerName"], "targetContainerAppName": img["targetContainerAppName"], "targetContainerAppEnvironmentName": img["targetContainerAppEnvironmentName"], "targetResourceGroup": img["targetResourceGroup"]})
+    info_list.append(inspect_result)
+
+
+def patch_interactive(cmd, resource_group_name=None, managed_env=None, show_all=False):
+    if is_docker_running() is False:
+        logger.warning("Please install or start Docker and try again.")
+        return
+    patchable_check_results = patch_list(cmd, resource_group_name, managed_env, show_all=show_all)
+    pack_exec_path = get_pack_exec_path()
+    if patchable_check_results is None:
+        return
+    patchable_check_results_json = json.dumps(patchable_check_results, indent=2)
+    without_unpatchable_results = []
```

Comment:
@snehapar9 @harryli0108 -- nit: do we need this line if we immediately instantiate it to values on the next line?

Reply:
I think this line is to handle the situation where the `patchable_check_results` is empty. This way, when it filters out the unpatchable images in next line, the variable `without_unpatchable_results` will remain to be empty.